### PR TITLE
feat: dependency-review action

### PR DESCRIPTION
**Pull request by mvayngrib**
adapted from https://github.com/ExodusMovement/exodus-hydra/pull/5426

test run: https://github.com/ExodusMovement/exodus-desktop/actions/runs/7729886611/job/21074067512?pr=15136
test run PR: https://github.com/ExodusMovement/exodus-desktop/pull/15136

ignore package.json "could not detect" warnings, it fails to detect anything that's not an exact version, but it still runs the check on the lockfiles
image

### CODE REVIEW

File: dependency-review/action.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+runs:
+  using: 'composite'
+  # timeout-minutes: 10 # TODO: https://github.com/actions/runner/issues/1979
+  steps:
+    - name: 'Checkout Repository'
+      uses: actions/checkout@1e31de5234b9f8995739874a8ce0492dc87873e2 # v4.0.0
+    - name: 'Dependency Review'
+      uses: actions/dependency-review-action@cc4f6536e38d1126c5e3b0683d469a14f23bfea4 # v3
+      with:
+        allow-licenses: MIT,Apache-2.0,ISC,BSD-2-Clause,BSD-3-Clause,Unlicense,CC0-1.0,CC-BY-3.0,WTFPL,0BSD,AFL-2.1
```
Comment (mvayngrib) [inline dependency-review/action.yml:12]: from here https://github.com/ExodusMovement/exodus-hydra/blob/b6189a4499d6103a799eca0c0779440344f5abee/tools/scripts/license-check/whitelists/licenses.js#L3-L16

**Comment by sparten11740** [comment]
on a second thought: do we need this here? This repo is for composite actions and the only deps we'll ever use here are dev deps

**Comment by mvayngrib** [comment]
the idea was to reuse this action in other repos to have a single list of licenses, not to verify deps in _this_ repo. Am i doing it wrong?

**Comment by sparten11740** [comment]
> the idea was to reuse this action in other repos to have a single list of licenses, not to verify deps in this repo. Am i doing it wrong?

ah gotcha, no you're not. my bad. Can we remove the comments? Otherwise good to go

**Comment by mvayngrib** [comment]
> Can we remove the comments?

done!

**Review by sparten11740 — APPROVED** [review]

**Pull request merged**